kubernetes-client-openapi/src/test/groovy/io/micronaut/kubernetes/client/openapi/KubernetesTokenAuthSpec.groovy

```diff
@@ -0,0 +1,119 @@
+package io.micronaut.kubernetes.client.openapi
+
+import io.micronaut.context.ApplicationContext
+import io.micronaut.context.annotation.Requires
+import io.micronaut.http.annotation.Controller
+import io.micronaut.http.annotation.Get
+import io.micronaut.http.annotation.Header
+import io.micronaut.kubernetes.client.openapi.api.CoreV1Api
+import io.micronaut.kubernetes.client.openapi.credential.model.ExecCredential
+import io.micronaut.kubernetes.client.openapi.credential.model.ExecCredentialStatus
+import io.micronaut.kubernetes.client.openapi.model.V1Pod
+import io.micronaut.kubernetes.client.openapi.model.V1PodList
+import io.micronaut.runtime.server.EmbeddedServer
+import spock.lang.AutoCleanup
+import spock.lang.Shared
+import spock.lang.Specification
+
+import java.nio.file.Files
+import java.nio.file.Path
+import java.time.ZoneId
+import java.time.ZonedDateTime
+
+class KubernetesTokenAuthSpec extends Specification {
+
+    static final String KUBE_CONFIG = """
+apiVersion: v1
+kind: Config
+clusters:
+  - name: test-cluster
+    cluster:
+      server: %s
+users:
+  - name: test-user
+    user:
+      exec:
+        apiVersion: client.authentication.k8s.io/v1beta1
+        command: curl
+        args:
+          - %s/credential
+        env:
+          - name: test-env-name
+            value: test-env-value
+contexts:
+  - name: test-context
+    context:
+      cluster: test-cluster
+      user: test-user
+current-context: test-context
+"""
+
+    @AutoCleanup
+    EmbeddedServer server = ApplicationContext.run(EmbeddedServer, [
+            'spec.name': 'TokenAuthServer',
+            'kubernetes.client.enabled': false
+    ])
+
+    @Shared
+    Path kubeConfigDir = Files.createTempDirectory("kube-temp-")
+
+    @Shared
+    Path kubeConfigFile = kubeConfigDir.resolve("config")
+
+    def cleanupSpec() {
+        if (kubeConfigFile != null) {
+            Files.deleteIfExists(kubeConfigFile)
+        }
+        if (kubeConfigDir) {
+            Files.deleteIfExists(kubeConfigDir)
+        }
+    }
+
+    def 'list pods when token authentication is used'() {
+        given:
+        kubeConfigFile.toFile().text = KUBE_CONFIG.formatted(server.URL, server.URL)
+        ApplicationContext clientContext = ApplicationContext.run([
+                'kubernetes.client.kube-config-path': "file:" + kubeConfigFile.toString()
+        ])
+
+        when:
+        V1PodList response = clientContext.getBean(CoreV1Api.class).listPodForAllNamespaces(
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null)
```

Comment:
wonder if there is a way to get the OpenAPI generator to generate overloaded variants that pass `null` to avoid having to write code like this?

Reply:
@andriy-dmytruk - do you maybe know answer to Graeme's above question?

Reply:
This is the only method signature that we generate. So far we just assume that calling a method is the simplest approach.

I guess we could generate another method that doesn't have any optional parameters. But if someone needs to set one optional parameter, they would need to use the full method and set all the others to null.

The OCI SDK client generates a separate class for each request that has a separate builder that allows ignoring all or some properties for the request. We could have an option to choose what to generate or even generate both. IMO that just complicates everything even more.

Reply:
we would need to generate 1 overloaded method per optional parameter and pass `null`. So in this case 11 methods would be generated. These could be `default` methods.

Reply:
Note I don't think this should block this PR but we should look to improve later